starknet_py/net/schemas/rpc.py

```diff
@@ -180,22 +200,57 @@ def make_dataclass(self, data, **kwargs) -> TransactionStatusResponse:
 
 
 class ResourcePriceSchema(Schema):
-    price_in_strk = Felt(data_key="price_in_strk", load_default=None)
+    price_in_fri = Felt(data_key="price_in_fri", load_default=None)
     price_in_wei = Felt(data_key="price_in_wei", required=True)
 
     @post_load
     def make_dataclass(self, data, **kwargs) -> ResourcePrice:
         return ResourcePrice(**data)
 
 
+class ResourceBoundsSchema(Schema):
+    max_amount = Felt(data_key="max_amount", required=True)
+    max_price_per_unit = Felt(data_key="max_price_per_unit", required=True)
+
+    @post_load
+    def make_dataclass(self, data, **kwargs) -> ResourceBounds:
+        return ResourceBounds(**data)
+
+
+class ResourceBoundsMappingSchema(Schema):
+    l1_gas = fields.Nested(ResourceBoundsSchema(), data_key="l1_gas", required=True)
+    l2_gas = fields.Nested(ResourceBoundsSchema(), data_key="l2_gas", required=True)
+
+    @post_load
+    def make_dataclass(self, data, **kwargs) -> ResourceBoundsMapping:
+        return ResourceBoundsMapping(**data)
+
+
 class TransactionSchema(Schema):
     hash = Felt(data_key="transaction_hash", load_default=None)
     signature = fields.List(Felt(), data_key="signature", load_default=[])
-    max_fee = Felt(data_key="max_fee", load_default=0)
     version = Felt(data_key="version", required=True)
 
 
-class InvokeTransactionSchema(TransactionSchema):
+class DeprecatedTransactionSchema(TransactionSchema):
+    max_fee = Felt(data_key="max_fee", required=True)
+
+
+class TransactionV3Schema(TransactionSchema):
+    tip = Felt(data_key="tip", load_default=0)
+    nonce_data_availability_mode = DAModeField(
+        data_key="nonce_data_availability_mode", load_default=DAMode.L1
+    )
+    fee_data_availability_mode = DAModeField(
+        data_key="fee_data_availability_mode", load_default=DAMode.L1
+    )
+    paymaster_data = fields.List(Felt(), data_key="paymaster_data", load_default=[])
+    resource_bounds = fields.Nested(
+        ResourceBoundsMappingSchema(), data_key="resource_bounds", required=True
+    )
+
+
+class DeprecatedInvokeTransactionSchema(DeprecatedTransactionSchema):
```

Comment:
I still think `InvokeTransactionSchemaV1` is a better name as it's more precise, is there a particular reason why you opted for `Deprecated` prefix?

Reply:
I agree, but we would need to split this into `InvokeTransactionV0Schema` and `InvokeTransactionV1Schema`. I am fine with doing this change, also for other deprecated schemas, but can it be addressed in #1249?

Reply:
Makes sense, you can do this in a separate PR if you prefer not to clutter this one